src/graphql/graphql-api.php

```diff
@@ -0,0 +1,245 @@
+			
+			// Flatten the inner blocks, if any.
+			if ( isset( $sourced_block['innerBlocks'] ) ) {
+				$sourced_block['innerBlocks'] = self::flatten_inner_blocks( $sourced_block['innerBlocks'], $filter_options );
+			}
+
+			// Convert the attributes to be in the name-value format that the schema expects.
+			if ( isset( $sourced_block['attributes'] ) && ! empty( $sourced_block['attributes'] ) && ! isset( $sourced_block['attributes'][0]['name'] ) ) {
+				$sourced_block['attributes'] = array_map(
+					function ( $name, $value ) {
+						return [
+							'name'  => $name,
+							'value' => $value,
+						];
+					},
+					array_keys( $sourced_block['attributes'] ),
+					array_values( $sourced_block['attributes'] )
+				);
+			}
+		}
+
+		return $sourced_block;
+	}
+
+	/**
+	 * Flatten the inner blocks, no matter how many levels of nesting is there.
+	 *
+	 * @param array $inner_blocks the inner blocks in the block.
+	 * @param array $filter_options Options to filter using, if any.
+	 * @param array $flattened_blocks the flattened blocks that's built up as we go through the inner blocks.
+	 * 
+	 * @return array
+	 */
+	public static function flatten_inner_blocks( $inner_blocks, $filter_options, $flattened_blocks = [] ) {
+		foreach ( $inner_blocks as $inner_block ) {
+			// This block doesnt have any inner blocks, so just add it to the flattened blocks. Ensure the parentId is set.
+			if ( ! isset( $inner_block['innerBlocks'] ) ) {
+				$inner_block['parentId'] = $inner_block['parentId'] ?? $filter_options['parentId'];
+				array_push( $flattened_blocks, $inner_block );
+				// This block is a root block, so go through the inner blocks recursively.
+			} elseif ( ! isset( $inner_block['parentId'] ) ) {
+				$inner_blocks_copy = $inner_block['innerBlocks'];
+				unset( $inner_block['innerBlocks'] );
+				$inner_block['parentId'] = $inner_block['parentId'] ?? $filter_options['parentId'];
+				array_push( $flattened_blocks, $inner_block );
+				$flattened_blocks = self::flatten_inner_blocks( $inner_blocks_copy, $filter_options, $flattened_blocks );
+			}
+		}
+
+		return $flattened_blocks;
+	}
+
+	/**
+	 * Register types and fields graphql integration.
+	 *
+	 * @return void
+	 */
+	public static function register_types() {
+		// Register the type corresponding to the attributes of each individual block.
+		register_graphql_object_type(
+			'BlockDataAttribute',
+			[
+				'description' => 'Block data attribute',
+				'fields'      => [
+					'name'  => [
+						'type'        => 'String',
+						'description' => 'Block data attribute name',
+					],
+					'value' => [
+						'type'        => 'String',
+						'description' => 'Block data attribute value',
+					],
+				],
+			],
+		);
+
+		// Register the type corresponding to the individual inner block, with the above attribute.
+		register_graphql_type(
+			'InnerBlockData',
+			[
+				'description' => 'Block data',
+				'fields'      => [
+					'id'         => [
+						'type'        => 'String',
```

Comment:
IDs in WPGraphQL are always non-nullable and are typically of type `ID` meaning a Relay ID. [Example](https://github.com/wp-graphql/wp-graphql/blob/develop/src/Type/InterfaceType/Node.php#L20). Create a Relay ID with `Relay::toGlobalId` [example](https://github.com/wp-graphql/wp-graphql/blob/1051357687339e53b7bd19b816c756316dbc8a47/src/Model/Menu.php#L72).
```suggestion
						'type'        => [ 'non_null' => 'ID' ],
```

Reply:
Good point, I have done this now for the ID and the parentId.